doozerlib/cli/release_gen_assembly.py

```diff
@@ -50,6 +66,31 @@ def exit_with_error(msg):
     if len(runtime.arches) != len(nightlies) + len(standards) and not custom:
         exit_with_error(f'Expected at least {len(runtime.arches)} nightlies; one for each group arch: {runtime.arches}')
 
+    if auto_previous and previous_list:
+        exit_with_error('Cannot use `--previous` and `--auto-previous` at the same time.')
+
+    if custom and (auto_previous or previous_list or in_flight):
+        exit_with_error("Custom release doesn't have previous list.")
+
+    # Calculate previous list
+    final_previous_list: Set[VersionInfo] = set()
+    if in_flight:
+        final_previous_list.add(VersionInfo.parse(in_flight))
+    if previous_list:
+        final_previous_list |= set(map(VersionInfo.parse, previous_list))
+    elif auto_previous:
+        # gen_assembly_name should be in the form of `fc.0`, `rc.1`, or `4.10.1`
+        if gen_assembly_name.startswith("fc") or gen_assembly_name.startswith("rc"):
+            major_minor = runtime.get_minor_version()  # x.y
+            version = f"{major_minor}.0-{gen_assembly_name}"
+        else:
+            version = gen_assembly_name
+        for arch in runtime.arches:
+            logger.info("Calculating previous list for %s", arch)
+            previous_list = util.get_release_calc_previous(version, arch, graph_url, graph_content_stable, graph_content_candidate, suggestions_url)
+            final_previous_list |= set(map(VersionInfo.parse, previous_list))
```

Comment:
Can you add a comment here explaining that we are merging for all arches - since individual arch difference doesn't matter - a note for future developers reading the code